doc-site/docs/architecture/noto.md

```diff
@@ -272,6 +376,171 @@ Inputs:
+    "name": "delegateLock",
+    "type": "function",
+    "inputs": [
+        {"name": "unlockHash", "type": "bytes32"},
+        {"name": "delegate", "type": "address"},
+        {"name": "signature", "type": "bytes"},
+        {"name": "data", "type": "bytes"}
+    ]
+}
+```
+
+Inputs:
+
+* **unlockHash** - EIP-712 hash of the prepared unlock, using type `Unlock(bytes32[] lockedInputs,bytes32[] lockedOutputs,bytes32[] outputs,bytes data)`
+* **delegate** - address of the delegate party that will be able to execute the unlock
+* **signature** - sender's signature (not verified on-chain, but can be verified by anyone with the private state data)
+* **data** - encoded Paladin and/or user data
+
+## Notary logic
+
+The notary logic (implemented in the domain [Go library](../../../domains/noto)) is responsible for validating and
+submitting all transactions to the base shared ledger.
+
+The notary will validate the following:
+
+- **Request Authenticity:** Each request to the notary will be accompanied by an EIP-712 signature from the sender,
+  which is validated by the notary. This prevents any identity from impersonating another when submitting requests.
+- **State Validity:** Each request will be accompanied by a proposed set of input and output UTXO states assembled
+  by the sending node. The notary checks that these states would be a valid expression of the requested operation -
+  for example, a "transfer" must be accompanied by inputs owned only by the "from" address, outputs owned by the
+  "to" address matching the desired transfer amount, and optionally some remainder outputs owned by the "from" address.
+  Note the distinction here of states that _would be_ valid - as final validation of spent/unspent state IDs will
+  be provided by the base ledger.
+- **Conservation of Value:** For most operations (other than mint and burn), the notary will ensure that the sum
+  of the inputs and of the outputs is equal.
+
+The above constraints cannot be altered without changing the library code. However, many other aspects of the
+notary logic can be easily configured as described below.
+
+### Notary mode: basic
+
+When a Noto contract is constructed with notary mode `basic`, the following notary behaviors can be configured:
+
+| Option         | Default | Description |
+| -------------- | ------- | ----------- |
+| restrictMint   | true    | _True:_ only the notary may mint<br>_False:_ any party may mint |
+| allowBurn      | true    | _True:_ token owners may burn their tokens<br>_False:_ tokens cannot be burned |
+| allowLock      | true    | _True:_ token owners may lock tokens (for purposes such as preparing or delegating transfers)<br>_False:_ tokens cannot be locked (not recommended, as it restricts the ability to incorporate tokens into swaps and other workflows) |
+| restrictUnlock | true    | _True:_ only the creator of a lock may unlock it<br>_False:_ any party may unlock (not recommended, as it allows anyone to claim locked tokens)
```

Comment:
may need to add some text on why _False_ is allowed at all, since it sounds unsafe.

Reply:
Yes, we can circle back on this. The main reason it is configurable in "basic" mode is because it is always off in "hooks" mode (so you must implement restrictions there if you want only the lock owner to unlock... but you can also do complex things to allow other parties to unlock if they provide some sort of other credentials as "data" to the unlock operation). I chose that anything which is "sometimes on and sometimes off" had a config flag.

We could make some of these things non-negotiable (always on) in "basic" mode. It probably just needs more explaining in terms of what things are always true in "basic" mode, what things are configurable in "basic" mode, and what things you must consider enforcing in "hooks" yourself because they're not enforced in the domain in "hooks" mode.

Reply:
Addressed in https://github.com/LF-Decentralized-Trust-labs/paladin/pull/596